src/v/config/tls_config.h

```diff
@@ -111,6 +115,12 @@ class tls_config {
         return std::nullopt;
     }
 
+    bool operator==(const tls_config& rhs) const {
+        return _enabled == rhs._enabled && _key_cert == rhs._key_cert
```

Comment:
`std::tie::operator==`?

Reply:
Converted these boilerplate equality functions to just use `= default` (thanks C++20!)